File: Cpp/src/leetcode/001/136-singleNumber.cpp
//
// Created by Administrator on 2024/10/22.
//
#include <algorithm>

#include "vector"

using namespace std;

class Solution {
public:
    int singleNumber_XOR(vector<int>& nums) {
        int result = 0;
        for (int num : nums) {
            result = result ^ num;
        }
        return result;
    }

    int singleNumber_OddEvenDigits(vector<int>& nums)
    {
        sort(nums.begin(), nums.end());
        int numa = 0;
        int numb = 0;
        for (int i = 0; i < nums.size() -1; ++i)
        {
            numa += nums[i];
            numb += nums[i + 1];
            i++;
        }
        numa += nums[nums.size() - 1];
        return numa - numb;
    }
};

File: Golang/Array/136-singleNumber.go
package Array

import "sort"

func singleNumber_XOR(nums []int) int {
	result := 0
	for _, num := range nums {
		result = result ^ num
	}
	return result
}

func singleNumber_OddEvenDigits(nums []int) int {
	if len(nums)%2 == 0 {
		return 0
	}

	sort.Ints(nums)
	numa := 0
	numb := 0
	println(numa, numb, len(nums))
	for i := 0; i < len(nums); i++ {
		numa += nums[i]
		numb += nums[i+1]
		i += 1
	}
	numa += nums[len(nums)-1]
	return numa - numb
}
